src/main/java/com/simondata/pouroversql/clients/ClientFactory.java

```diff
@@ -72,4 +72,28 @@ public static SQLClient makeSQLClient(SqlEngine engine, SQLParams params) {
         }
         return client;
     }
+
+    public static SFTPClient makeSFTPClient(String sftpType, SFTPParams params) {
```

Comment:
I generally get confused when classes have >1 method with the same name, or even methods with similar names that do similar things. If possible, it's best to reduce the publicly exposed methods to just what a user might want. Make everything else private. In this case, you could probably just have a single method. The same feedback applies to the overloaded methods in the other classes.

Reply:
Sounds good, that makes sense. Went ahead and cleaned up those methods all over

Reply:
One follow-up to that; SQLExtractor already existed, so should I leave those methods relatively untouched? Had gone through and cleaned them up, but not sure if I should on those.